File: src/main/java/com/mallang/auth/config/OauthWebConfig.java
package com.mallang.auth.config;

import com.mallang.auth.presentation.support.OauthServerTypeConverter;
import org.springframework.context.annotation.Configuration;
import org.springframework.format.FormatterRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@Configuration
public class OauthWebConfig implements WebMvcConfigurer {

    @Override
    public void addFormatters(FormatterRegistry registry) {
        registry.addConverter(new OauthServerTypeConverter());
    }
}

File: src/main/kotlin/com/mallang/auth/config/OauthWebConfig.kt
package com.mallang.auth.config

import com.mallang.auth.presentation.support.OauthServerTypeConverter
import org.springframework.context.annotation.Configuration
import org.springframework.format.FormatterRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
class OauthWebConfig : WebMvcConfigurer {
    
    override fun addFormatters(registry: FormatterRegistry) {
        registry.addConverter(OauthServerTypeConverter())
    }
}
